apps/design/backend/src/worker/tasks.ts

```diff
@@ -0,0 +1,68 @@
+import fs from 'fs/promises';
+import JsZip from 'jszip';
+import path from 'path';
+import { z } from 'zod';
+import { throwIllegalValue } from '@votingworks/basics';
+import { layOutAllBallotStyles } from '@votingworks/hmpb-layout';
+import {
+  BallotType,
+  ElectionPackageFileName,
+  getDisplayElectionHash,
+  Id,
+  safeParseJson,
+} from '@votingworks/types';
+
+import { PORT } from '../globals';
+import { BackgroundTask } from '../store';
+import { Workspace } from '../workspace';
+
+async function generateElectionPackage(
+  { assetDirectoryPath, store }: Workspace,
+  { electionId }: { electionId: Id }
+): Promise<void> {
+  const { election, layoutOptions, systemSettings } =
+    store.getElection(electionId);
+  const { electionDefinition } = layOutAllBallotStyles({
+    election,
+    // Ballot type and ballot mode shouldn't change the election definition, so it doesn't matter
+    // what we pass here
+    ballotType: BallotType.Precinct,
+    ballotMode: 'test',
+    layoutOptions,
+  }).unsafeUnwrap();
+  const zip = new JsZip();
+  zip.file(ElectionPackageFileName.ELECTION, electionDefinition.electionData);
+  zip.file(
+    ElectionPackageFileName.SYSTEM_SETTINGS,
+    JSON.stringify(systemSettings, null, 2)
+  );
+
+  const displayElectionHash = getDisplayElectionHash(electionDefinition);
+  const fileName = `election-package-${displayElectionHash}.zip`;
+  const filePath = path.join(assetDirectoryPath, fileName);
+  await fs.writeFile(filePath, await zip.generateAsync({ type: 'nodebuffer' }));
```

Comment:
Good call! https://github.com/votingworks/vxsuite/pull/4462/commits/502a424feb46465ce4cfba169718e6e58eba3482

It does look like the `streamFiles: true` option comes with some risks, but since we're importing election packages in such controlled contexts, we should be okay.

<table><img width="870" alt="note" src="https://github.com/votingworks/vxsuite/assets/12616928/b9af167f-d58e-4dcb-afcb-9df351de9337"></table>

[Reference](https://stuk.github.io/jszip/documentation/api_jszip/generate_async.html#streamfiles-option)